hw/ip/rv_dm/data/rv_dm.hjson

```diff
@@ -103,15 +103,146 @@
     regs:[
     ]
     mem: [
-      // Debug memory size (given as `items` below) must be a power of two.
+      // Expansion of debug memory within rv_dm used for DV purposes. The generated reg_top is not
+      // actually instantiated in RTL since the debug memory region is implemented inside the
+      // vendored-in module from the PULP project.
+      //
+      // The debug memory region is specified in the PULP debug system documentation here:
+      // https://github.com/pulp-platform/riscv-dbg/blob/master/doc/debug-system.md
+      //
+      // The debug module exposes a 16kB memory called debug memory. It has a ROM portion
+      // (debug ROM), some memory mapped CSRs and a RAM portion (program buffer). This region is
+      // accessible over the TL interface only if debug mode is active.
+      { skipto: "0x100" }
+      { multireg: {
+          cname: "HALTED"
+          name:  "HALTED"
+          desc: "Write to this address to acknowledge that the core has halted.",
+          count: "NrHarts"
```

Comment:
does `NrHarts` long term need to be templated like in `rv_plic`?

Reply:
you mean to feed it in from the topgen configuration? probably yes - although it would be nice if we found a way to pass such top-level params into the hjson without having to make the entire IP a generated IP (e.g. if only the hjson needs modifying, but the RTL does not).

Reply:
yeah, but this kind of feeds back to the age old parmaeterized reg problem that we have.
I think we still don't have a good solution for that outside of templating. These parameters would technically show up in `rv_dm_reg_pkg.sv` or something right?

Reply:
yeah that's right...

Reply:
problem for another day :)